hub/api/tensor.py

```diff
@@ -0,0 +1,44 @@
+from hub.util.slice import merge_slices
+import numpy as np
+from typing import Union
+
+
+class Tensor:
```

Comment:
Do u think we can add other things that the Tensor is aware of like metadata etc as discussed in this class?